shell/platform/android/io/flutter/view/FlutterView.java

```diff
@@ -173,6 +175,8 @@ public void surfaceDestroyed(SurfaceHolder holder) {
             StringCodec.INSTANCE);
         mFlutterSystemChannel = new BasicMessageChannel<>(this, "flutter/system",
             JSONMessageCodec.INSTANCE);
+        mFlutterSettingsChannel = new BasicMessageChannel<>(this, "flutter/settings",
```

Comment:
The differences between a "system" channel and a "settings" channel are not immediately obvious. So why a new channel? You can disambiguate by the setter by using the type already.

Reply:
In a separate patch probably, we should also look into moving away from the JSON message codec.

Reply:
We support multiple channel precisely so we can send different types of data over different channels. There's also significant semantic differences between user settings and other kinds of system messages (currently only memory pressure). User settings are passed through the method channel _before_ we call `main()` over on the Dart side, then again when the values are updated. IOW they are not "events" but "values changing over time". That's not the case for system events, such as "memory pressure". Those are truly events. They are not persistent values. Take a look at extra logic in `HandleSystemPlatformMessage` just to deal with this distinction.